docs/using-lexical.mdx

```diff
@@ -0,0 +1,49 @@
+import { Meta } from "@storybook/blocks";
+
+<Meta title="Documentation/Using Lexical" />
+
+# Lexical
+
+## Why Lexical?
+
+In previous versions of Carbon, the `Note` and `TextEditor` components were built using [DraftJS](https://draftjs.org/), a framework developed by Facebook. However, `DraftJS` has been deprecated and is no longer maintained. As a result, we have migrated to a new  framework called
+[Lexical](https://lexical.dev), which is also developed by Meta. In doing so, we have been able to maintain the functionality of the `Note` and `TextEditor` components while also ensuring that they are up-to-date and well-maintained. `Lexical` is a powerful and flexible framework
+that allows for the creation of rich text editors and other text-based components, and is fully customisable which allows us to cater for a wide range of use cases.
+
+### Packages
+
+The following `lexical` packages are used in Carbon to provide the functionality for `Note` and `TextEditor`:
+
+- `lexical`: The core package;
+- `@lexical/react`: A wrapper around the core package for React;
+- `@lexical/headless`: A headless version of the core package used internally for testing the components;
+- `@lexical/link`: A package for handling links in the `TextEditor`;
+- `@lexical/selection`: A package for handling selections in the `TextEditor`.
+
+### Usage
+
+First, import the relevant components from the `carbon-react` package:
+
+```jsx
+import TextEditor, { CreateFromHTML } from 'carbon-react/lib/components/text-editor';
```

Comment:
nitpick: for consistency we could name these in camel-case format as they are functions rather than functional components etc

`createEmpty` and `createFromHTML` etc